test/e2e/specs/editor/blocks/table.spec.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Table', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'displays a form for choosing the row and column count of the table', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+
+		// Check for existence of the column count field.
+		const columnCountInput = page.locator(
+			'role=spinbutton[name="Column count"i]'
+		);
+		await expect( columnCountInput ).toBeVisible();
+
+		// Modify the column count.
+		await columnCountInput.click();
+		await expect( columnCountInput ).toHaveValue( '2' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.type( '5' );
+
+		// Check for existence of the row count field.
+		const rowCountInput = page.locator(
+			'role=spinbutton[name="Row count"i]'
+		);
+		await expect( rowCountInput ).toBeVisible();
+
+		// Modify the row count.
+		await rowCountInput.click();
+		await expect( rowCountInput ).toHaveValue( '2' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.type( '10' );
+
+		// Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Expect the post content to have a correctly sized table.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'allows text to by typed into cells', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+
+		// Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Click the first cell and add some text.
+		await page.click( 'role=textbox[name="Body cell text"] >> nth=0' );
+		await page.keyboard.type( 'This' );
+
+		// Navigate to the next cell and add some text.
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'is' );
+
+		// Navigate to the next cell and add some text.
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'table' );
+
+		// Navigate to the next cell and add some text.
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'block' );
+
+		// Expect the post to have the correct written content inside the table.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'allows header and footer rows to be switched on and off', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+		await editor.openDocumentSettingsSidebar();
+
+		const headerSwitchSelector = 'role=checkbox[name="Header section"i]';
+		const footerSwitchSelector = 'role=checkbox[name="Footer section"i]';
```

Comment:
(This is plain nitpicking:) With Playwright, it's preferred to create variables for the locator instances than for the selector strings. Locators are lazy, and they can better represent the elements themselves.

So instead of passing strings around, we can just invoke methods on the locators instead:

```js
const headerSwitch = page.locator( 'role=checkbox[name="Header section"i]' );

await expect( headerSwitch ).toBeHidden();
await expect( headerSwitch ).toBeVisible();
await headerSwitch.check();
```

This creates slightly more readable code IMO :)